hooks/useConditionalTokens.ts

```diff
@@ -47,53 +48,53 @@ export default function useConditionalTokens() {
     fetchAmount: fetchUnderlyingMeta,
   } = useBalance(base.underlyingTokenMint);
   const {
-      amount: pMetaBalance,
-      fetchAmount: fetchPassMeta,
+    amount: pMetaBalance,
+    fetchAmount: fetchPassMeta,
   } = useBalance(base.conditionalOnFinalizeTokenMint);
   const {
-      amount: fMetaBalance,
-      fetchAmount: fetchFailMeta,
+    amount: fMetaBalance,
+    fetchAmount: fetchFailMeta,
   } = useBalance(base.conditionalOnRevertTokenMint);
 
   const {
-      amount: pUsdcBalance,
-      fetchAmount: fetchPassUsdc,
+    amount: pUsdcBalance,
+    fetchAmount: fetchPassUsdc,
   } = useBalance(quote.conditionalOnFinalizeTokenMint);
   const {
-      amount: fUsdcBalance,
-      fetchAmount: fetchFailUsdc,
+    amount: fUsdcBalance,
+    fetchAmount: fetchFailUsdc,
   } = useBalance(quote.conditionalOnRevertTokenMint);
 
   const [metaToken, setMetaToken] = useState<ConditionalToken | undefined>();
   const [usdcToken, setUsdcToken] = useState<ConditionalToken | undefined>();
 
   useEffect(() => {
-      if (tokens && base && quote) {
-          setMetaToken({
-              token: tokens.meta as unknown as Token,
-              symbol: tokens.meta?.symbol as unknown as string,
-              balanceSpot: metaBalance,
-              balancePass: pMetaBalance,
-              balanceFail: fMetaBalance,
-              fetchUnderlying: fetchUnderlyingMeta,
-              fetchPass: fetchPassMeta,
-              fetchFail: fetchFailMeta,
-              finalize: base.conditionalOnFinalizeTokenMint,
-              revert: base.conditionalOnRevertTokenMint,
-          });
-          setUsdcToken({
-              token: tokens.usdc as unknown as Token,
-              symbol: tokens.usdc?.symbol as unknown as string,
-              balanceSpot: usdcBalance,
-              balancePass: pUsdcBalance,
-              balanceFail: fUsdcBalance,
-              fetchUnderlying: fetchUnderlyingUsdc,
-              fetchPass: fetchPassUsdc,
-              fetchFail: fetchFailUsdc,
-              finalize: quote.conditionalOnFinalizeTokenMint,
-              revert: quote.conditionalOnRevertTokenMint,
-          });
-      }
+    if (tokens && base && quote) {
+      setMetaToken({
```

Comment:
One idea is to pass this function into the parsing of WS message so we end up setting all the balances as necessary.

Reply:
oh yeah I think that is an area that makes a lot of sense to do.